packages/ember-routing/tests/system/router_test.js

```diff
@@ -261,5 +261,20 @@ moduleFor('Ember Router', class extends AbstractTestCase {
 
     triggerEvent(handlerInfos, false, ['loading']);
   }
-});
 
+  ['@test transitionTo should throw an error when called after owner is destroyed'](assert) {
+    let router = createRouter();
+
+    runDestroy(owner);
+
+    router.currentRouteName = 'route-a';
+
+    assert.throws(function() {
```

Comment:
Since assertions are stripped in production builds (and we run tests against production builds) this will need to be tweaked. The main update should be to replace `assert.throws` with `expectAssertion` (which is a global).